File: Intensive.Web/src/app/aric/aric-target.ts
export class AricTarget {
    href: string;
    rel: string;


    constructor(){ }
}

export class AricCloudTarget extends AricTarget {
   
    constructor(dc:string, tenant: number, instance: string){
        super();
        this.href = "https://<CloudDC>.servers.api.rackspacecloud.com/v2/<tenant>/servers/<InstanceID>";
        this.rel = "http://schemas.automation.rackspacecloud.com/targets/device";

    }
}

export class AricDedicatedTarget extends AricTarget {

    constructor(device: number){
        super();
        this.href = "http://core.rackspace.com/py/core/#/device/"+ device;
        this.rel = "http://schemas.automation.rackspacecloud.com/targets/device"
    }
}

File: Intensive.Services/ARIC/AricTarget.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Intensive.Services.Aric
{
    public class AricTarget
    {
        public string href { get; set; }
        public string rel { get; set; }

        public AricTarget() { }

        public AricTarget(int deviceNumber)
        {
            this.href = $"http://core.rackspace.com/py/core/#/device/{deviceNumber}";
            this.rel = "http://schemas.automation.rackspacecloud.com/targets/device";
        }
        public AricTarget(string dc, int tenant, string instance)
        {
            this.href = $"https://{dc}.servers.api.rackspacecloud.com/v2/{tenant}/servers/{instance}";
            this.rel = "http://schemas.automation.rackspacecloud.com/targets/device";
        }
    }
}
